src/core/meshcop/tcat_agent.hpp

```diff
@@ -345,7 +371,8 @@ class TcatAgent : public InstanceLocator, private NonCopyable
     ExtendedPanId                    mCommissionerExtendedPanId;
     char                             mCurrentServiceName[OT_TCAT_MAX_SERVICE_NAME_LENGTH + 1];
     State                            mState;
-    bool                             mAlreadyCommissioned : 1;
+    uint8_t                          mAdvertisement[OT_TCAT_ADVERTISEMENT_MAX_LEN];
+    uint8_t                          mAdvertisementLength;
```

Comment:
If I am not mistaken, the `mAdvertisement` buffer is only used to return the generated byte sequence from `GetAdvertisementData()`. Do we need to retain this as a member variable in `TcatAgent`?

We can consider changing `GetAdvertisementData()` for the caller to pass the buffer where the generated sequence is placed?

Reply:
now when I think about it I can change this in the folllowing way:
1. remove mAdvertisement.
2. add new platform API to get preallocated advertisement buffer (in zephyr case such has to exist for BLE advertisement).
3. Emplace the data on the buffer.

Reply:
Sounds reasonable to me.